code/espurna/sensors/BME680Sensor.h

```diff
@@ -0,0 +1,241 @@
+// -----------------------------------------------------------------------------
+// BME680 Sensor over I2C
+// Copyright (C) 2017-2019 by Xose Pérez <xose dot perez at gmail dot com>
+// -----------------------------------------------------------------------------
+
+#if SENSOR_SUPPORT && BME680_SUPPORT
+
+#pragma once
+
+#include "Arduino.h"
+#include "I2CSensor.h"
+#include "bsec.h"
+
+// The maximum allowed time between two `bsec_sensor_control` calls depends on
+// configuration profile `bsec_config_iaq` below. With these settings, the
+// maximum allowed time is 3s.
+#define SENSOR_READ_INTERVAL 1
```

Comment:
@mcspr I don't really like this approach.. do you have better suggestions?

Reply:
'Maximum allowed time' meaning we **must** call it at least every 3 seconds?
There is always tick(), which gets called independently of the reading and can be polled via `millis() - last >= timeout_milliseconds` / `micros() - last >= timeout_microseconds`

Reply:
Correct. In Low Power (LP) mode, which is the default, a reading *must* happen at least every 3 seconds.

`tick()` is exactly what I was looking for. I'll make this modification.

Reply:
https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bme680-ds001.pdf
> 4.1 BSEC Software
> ...
> Low power (LP) mode that is designed for interactive applications where the indoor-air-quality is tracked and
observed at a higher update rate of 3 seconds with a current consumption of <1 mA

Technically, we just loose last 3 seconds or readings if we don't? It is still useful though, as we don't have any new data with less than 3 seconds.

Reply:
From my observations, if we don't call within 3 seconds, the sensor can never calibrate. The default was at 6 seconds and the readings were stalled. I believe it's part of the algorithm.